addons/buildables/XEH_preInit.sqf

```diff
@@ -0,0 +1,80 @@
+#include "script_component.hpp"
+ADDON = false;
+
+PREP_RECOMPILE_START;
+#include "XEH_PREP.hpp"
+PREP_RECOMPILE_END;
+
+GVAR(areas) = createHashMap;
+
+[QGVAR(registerArea), {
+    TRACE_1("Registering area",_this);
+
+    GVAR(areas) set [_this get AREA_ID_KEY, _this];
+
+    {
+        _x call FUNC(addSupplyDeliverAction);
+    } forEach (_this get "supplyCrates");
+}] call CBA_fnc_addEventHandler;
+
+if (hasInterface) then {
+    GVAR(intializedClasses) = [];
+
+    [QGVAR(deliverNotify), {
+        params ["_built"];
+
+        if (_built) exitWith {
+            "Built" call CBA_fnc_notify;
+            playSound "defaultNotification";
+        };
+
+        "Delivered" call CBA_fnc_notify;
+        playSound "defaultNotificationClose";
+    }] call CBA_fnc_addEventHandler;
+};
+
+if (isServer) then {
+    [QGVAR(deliverSupply), {
+        params ["_areaId", ["_supply", objNull], ["_supplier", [objNull]]];
+
+        private _area = GVAR(areas) get _areaId;
+
+        INFO_3("Supply delivered to area - %1, %2, %3",_area get "displayName",_areaId,_supply);
+
+        private _required = _area get "requiredSupplies";
+        private _current = _area get "currentSupplies";
+        _current = _current + 1;
+
+        _area set ["currentSupplies", _current];
+
+        deleteVehicle _supply;
```

Comment:
We don't check if `_supply` is null while, is this expected behavior? Just wondering.

Reply:
Yes, we allow delivering "virtual" supplies if someone wants to script it.